src/core/flow.js

```diff
@@ -455,9 +466,13 @@ export function FlowWithPhases({
     // Update undo / redo state.
     if (!endTurn) {
       const undo = state._undo || [];
+      const moveType =
+        action !== undefined && action.payload !== undefined
```

Comment:
When is `action` ever `undefined`?

Reply:
Basically both checks are there because of failing tests, for me it looks like this wouldn't happen 'in production', but I am not sure and didn't want to mess with other tests.
Should I update the failing test(s) instead?

    FAIL  src/core/flow.test.js
    ● onMove
    
    TypeError: Cannot read property 'payload' of undefined
      
      at processMove (src/core/flow.js:469:29)
      at Object.processMove (src/core/flow.js:85:14)
      at Object.<anonymous> (src/core/flow.test.js:190:18)
          at new Promise (<anonymous>)
          at <anonymous>

Reply:
so since [#151](https://github.com/google/boardgame.io/pull/151/files) also assumes action and payload to be there, I just fixed the test, by giving them what they needed. If it's not what you're expecting, I will change it...